runner/src/index.ts

```diff
@@ -55,3 +57,61 @@ export function test() {
 
   return 0;
 }
+
+const matchIdenticalTopLevel = (output: any) => {
+  // determine top-level fields that should be identical
+  // NOTE: anEmbeddedObject, anEmbeddedObjectArray, and aDate are intentionally omitted here
+  const matchIdentical = [
+    "anOptionalString",
+    "aString",
+    "anInt",
+    "aFloat",
+    "aDouble",
+    "anUntypedObject",
+    "anEnum",
+  ] as const;
+  matchIdentical.forEach((k: typeof matchIdentical[number]) => {
+    let key: keyof typeof KitchenSink = k;
+    let actual = output[key];
+    let expected = KitchenSink[key];
+    if (key === "anUntypedObject") {
+      actual = JSON.stringify(actual);
+      expected = JSON.stringify(expected);
+    }
+    Test.assertEqual(
+      `reflectJsonObject preserved identical value '${key}'`,
+      actual,
+      expected,
+    );
+  });
+};
+
+const matchIdenticalEmbedded = (embedded: any) => {
+  // determine flat embedded items that should be identical
+  // NOTE: aDate is intentionally omitted here
+  const matchIdenticalEmbedded = [
+    "aBoolArray",
+    "aStringArray",
+    "anEnumArray",
+    "anIntArray",
+  ] as const;
+  matchIdenticalEmbedded.forEach((k: typeof matchIdenticalEmbedded[number]) => {
+    let key: keyof typeof EmbeddedObject = k;
+    Test.assertEqual(
+      `reflectJsonObject preserved identical value '${key}'`,
+      JSON.stringify(embedded[k]),
+      JSON.stringify(EmbeddedObject[key]),
```

Comment:
@bhelx - how do you feel about this? I re-encode these values before testing them. without doing so, they "appear" equal, but fail this test (which does a `===` comparison). 

for example, before re-encoding them: 

```
FAIL ...... reflectJsonObject preserved identical value 'aStringArray'
  Expected Hello,🌍,World! === Hello,🌍,World!
    at assertEqual (script.js:132)
    at <anonymous> (script.js:243)
    at forEach (native)
    at matchIdenticalEmbedded (script.js:244)
    at forEach (native)
    at test (script.js:193)
```

Reply:
What is the difference? 😆

Reply:
exactly! idk!

Reply:
oh I know... `===` on Objects and Arrays (maybe more) checks that the operands are the same instance not just the same type and value. 

so I think it's actually recommended to stringify them before comparing.